pkg/pod/entrypoint_lookup_impl_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	remotetest "github.com/tektoncd/pipeline/test"
+	"github.com/tektoncd/pipeline/test/parse"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	username = "foo"
+	password = "bar"
+)
+
+type fakeHTTP struct {
+	reg http.Handler
+}
+
+func newfakeHTTP() fakeHTTP {
+	reg := registry.New()
+	return fakeHTTP{
+		reg: reg,
+	}
+}
+
+func (f *fakeHTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Requests to container registry except "GET" will be handled by the registry handler.
+	if r.Header["User-Agent"][0] == "go-containerregistry" && !(strings.HasPrefix(r.URL.Path, "/v2/task") && r.Method == "GET") {
+		f.reg.ServeHTTP(w, r)
+		return
+	}
+
+	// For Ping requests, the server will need basic authentication and reutrn 401
+	if r.Header["Authorization"] == nil {
+		w.Header().Add("WWW-Authenticate", "basic")
+		w.WriteHeader(401)
+	} else {
+		// Handle the requests with matching authorization.
+		auth := strings.Split(r.Header["Authorization"][0], " ")[1]
+		if auth == base64.StdEncoding.EncodeToString([]byte(username+":"+password)) {
+			f.reg.ServeHTTP(w, r)
+		} else {
+			w.WriteHeader(401)
+		}
+
+	}
+}
+
+func TestGet(t *testing.T) {
+
+	for _, tc := range []struct {
+		testname string
+		username string
+		password string
+	}{{
+		testname: "correct auth",
+		username: "foo",
+		password: "bar",
+	},
+		{
+			testname: "empty auth",
+			username: "",
+			password: "",
+		},
+	} {
```

Comment:
Might also be worth testing a secret that exists but isn't authorized, no secret (in case empty string is different than nil), and a secret that does exist at all.